### PR TITLE
Refactor/Proj 92

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
**This PR introduces several improvements and bug fixes to the Sign Up component.**

### **Form Validation Enhancements:**

Added validation for the userName field to ensure it is between 6 and 30 characters long and matches the required format.
Strengthened password validation to include special characters and numbers, with a minimum length of 6 and a maximum of 20 characters.
Implemented a refine check to ensure the confirmPassword matches the password field.
Added a checkbox for agreeing to terms and conditions, with validation to ensure it is checked.
### 
**Global Error Handling:**

Improved error handling to process global errors from the server. If the server returns a bad request, the form fields will be updated with the corresponding error messages.
Added toast notifications for successful sign-up, server errors, and other unexpected errors.

### **Modal Notification for Email:**

Introduced a modal component (SentEmail) that appears after a successful sign-up, displaying a confirmation message and the user's email address.
The modal includes a close button to dismiss the notification.

### **Form Reset:**

Upon successful sign-up or closing the modal, the form is reset to its initial state.

### **UI Improvements:**

Styling enhancements for better visual presentation, including input fields, checkboxes, buttons, and error messages.
### 
**Code Refactoring:**

Extracted the form logic into a separate custom hook (useSignUpForm) to improve reusability and maintainability.
These changes aim to enhance the user experience by providing better form validation, error handling, and feedback. The component now also includes a modal notification for successful sign-ups, improving user awareness of the registration process.

### CODE REVIEW

File: src/features/auth/ui/sentEmail/sentEmail.tsx [hunk 1]
```diff
@@ -14,8 +15,9 @@ import styles from './sentEmail.module.scss'
 export type SentEmailProps = {
   closeModal: () => void
   email?: string
+  open: boolean
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.tsx:18]: maybe isOpen

File: src/features/auth/ui/signUp/signUp.tsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { GithubSvgrepoCom31, GoogleSvgrepoCom1 } from '@/assets'
+import { SentEmail } from '@/features'
 import { useSignUpForm } from '@/features/auth/ui/signUp/useSignUpForm'
```
Comment (Lelik7777) [inline src/features/auth/ui/signUp/signUp.tsx]: ./useSignUpForm

File: src/features/auth/ui/signUp/signUp.tsx [hunk 3]
```diff
@@ -1,3 +1,5 @@
...
 import { GithubSvgrepoCom31, GoogleSvgrepoCom1 } from '@/shared/assets/icons'
```
Comment (Lelik7777) [inline src/features/auth/ui/signUp/signUp.tsx]: дублирование имортов

File: src/features/auth/ui/signUp/useSignUpForm.tsx [hunk 4]
```diff
@@ -36,34 +37,79 @@ const signUpSchema = z
...
     resolver: zodResolver(signUpSchema),
   })
 
-  const onSubmit = (data: FormValues) => {
-    //toDo: connect to api
-    setUserEmail(data.email)
-    setIsModalOpen(true)
-    reset()
-  }
+  const onSubmit = handleSubmit(data => {
```
Comment (Lelik7777) [inline src/features/auth/ui/signUp/useSignUpForm.tsx:72]: type data?

File: src/features/auth/ui/signUp/useSignUpForm.tsx [hunk 5]
```diff
@@ -36,34 +37,79 @@ const signUpSchema = z
...
+    console.log('Data to be sent:', data)
+    signUp(data)
```
Comment (Lelik7777) [inline src/features/auth/ui/signUp/useSignUpForm.tsx:74]: может через async await and try catch ?

File: src/pages/auth/signUp/index.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { SignUp } from '@/features/auth/ui/signUp'
```
Comment (Lelik7777) [inline src/pages/auth/signUp/index.tsx]: import { SignUp,getAuthLayout } from '@/features

File: src/features/auth/ui/sentEmail/sentEmail.tsx (hunk 1 above)
Comment (EvgeniyaYakunina) [reply src/features/auth/ui/sentEmail/sentEmail.tsx:18]: в библиотеке Radix это свойство под названием open. Решила оставить его для однообразия модалок.

File: src/features/auth/ui/signUp/useSignUpForm.tsx (hunk 4 above)
Comment (EvgeniyaYakunina) [reply src/features/auth/ui/signUp/useSignUpForm.tsx:72]: Ищу вариант как написать эту функцию

File: src/features/auth/ui/signUp/useSignUpForm.tsx (hunk 5 above)
Comment (EvgeniyaYakunina) [reply src/features/auth/ui/signUp/useSignUpForm.tsx:74]: В конечном итоге постараюсь так и сделать.

**Pull request merged**